server/src/main/java/io/crate/protocols/postgres/KeyData.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Licensed to Crate.io GmbH ("Crate") under one or more contributor
+ * license agreements.  See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.  Crate licenses
+ * this file to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial agreement.
+ */
+
+package io.crate.protocols.postgres;
+
+import io.netty.buffer.ByteBuf;
+
+import java.util.concurrent.ThreadLocalRandom;
+import java.util.concurrent.atomic.AtomicInteger;
+
+record KeyData(int pid, int secretKey) {
+
+    static KeyData generate(AtomicInteger startingPID) {
+        int secretKey = ThreadLocalRandom.current().nextInt();
+        return new KeyData(startingPID.incrementAndGet(), secretKey);
```

Comment:
```suggestion
    static KeyData generate(int pid) {
        int secretKey = ThreadLocalRandom.current().nextInt();
        return new KeyData(pid, secretKey);
```

I'd do the incrementAndGet call outside. Otherwise it is not clear without looking at the implementation what the `KeyData` is doing with the AtomicInteger and whether it just gets the current number or whether it increments it.